### ISSUE TITLE
Style/SymbolProc false positive with unused argument

### ISSUE THREAD

**Issue by vlad-pisanov** [rubocop-hq/rubocop#9257]
Consider:

```ruby
cat = Cat.new
[1, 2, 3].map { |_i| cat.meow }
```

## Expected behavior

Should not register any offenses.

## Actual behavior

Rubocop registers Style/SymbolProc offense: `[Correctable] Style/SymbolProc: Pass &:meow as an argument to map instead of a block.`

With the `-A` option, it autocorrects to:

```ruby
cat = Cat.new
[1, 2, 3].map(&:meow)
```

## RuboCop version

NOTE: this appears to be a regression because version 1.4.2 doesn't exhibit this behavior. Likely related to: https://github.com/rubocop-hq/rubocop/pull/9232

```
1.6.1 (using Parser 2.7.2.0, rubocop-ast 1.3.0, running on ruby 2.4.5 x86_64-darwin19)
  - rubocop-minitest 0.10.1
  - rubocop-performance 1.9.1
  - rubocop-rails 2.9.1
```

**Comment by marcandre** [comment]
Thanks for the report.
I don't think this was caused by #9232, but maybe by #9127.
@dvandersluis would you like to check this out?

**Comment by dvandersluis** [comment]
Sure thing!
